msf/preflight.go

```diff
@@ -344,6 +335,14 @@ func _preflight(readerFn func(callback gcode.LineCallback) error, palette *Palet
 		if layerTopZs := len(results.layerTopZs); layerTopZs != results.totalLayers {
 			return results, fmt.Errorf("invariant violation: expected %d layerTopZs, got %d", results.totalLayers, layerTopZs)
 		}
+		for i := 0; i < results.totalLayers; i++ {
+			if results.layerThicknesses[i] <= 0 {
+				return results, fmt.Errorf("invariant violation: zero thickness at layer %d", i)
+			}
+			if results.layerTopZs[i] == 0 {
+				return results, fmt.Errorf("invariant violation: zero height at layer %d", i)
+			}
```

Comment:
This matches what we have on dev but based on previous discussions, will this cause a problem if our Z-offset matches the first layer thickness such that the first layer's top Z is 0? I'm good to keep this as-is if we can test that case and it works as expected, otherwise we may need to do a more specific check here to say something like the following instead:

```golang
if i > 0 && !(results.layerTopZs[i] > results.layerTopZs[i-1]) {}
```